tools/project-context.js

```diff
@@ -985,7 +1013,7 @@ _.extend(exports.ProjectConstraintsFile.prototype, {
   eachConstraint: function (iterator) {
     var self = this;
     _.each(self._constraintLines, function (lineRecord) {
-      if (lineRecord.constraint)
+      if (! lineRecord.skipOnRead && lineRecord.constraint)
```

Comment:
> Feels like that might be overkill but I'm not too familiar with this part of the code.

Yes, that would be unnecessarily complex solution.

Another option would be to iterate over values of `self._constraintMap`. But I'm not sure if `eachConstraint` must iterate with correct order (first-in, first-out) and if so whether this would be still guaranteed.